### PR TITLE
{Service Connector} Some improvements on command experience

### PR DESCRIPTION
**Pull request by houk-ms**
**Related command**

`az spring connection list-support-types`
`az {source} connection validate`

**Description**

This PR fixes the following issues
- `az spring connection list-support-types` doesn't work

The PR also provides the following improvements
- Support customizing auth info when using `--new` parameter 
- `az {source} connection validate` output improvement, only show useful info

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py [hunk 1]
```diff
@@ -1083,6 +1083,15 @@ class CLIENT_TYPE(Enum):
...
         ],
+        RESOURCE.SignalR: [
+            CLIENT_TYPE.Dotnet,
+            CLIENT_TYPE.Blank
+        ],
+        RESOURCE.WebPubSub: [
+            CLIENT_TYPE.Java,
+            CLIENT_TYPE.Dotnet,
+            CLIENT_TYPE.Blank
+        ],
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:1094]: @xfz11 please confirm the clientType here

**Comment by yonzhan** [comment]
Service Connector

**Review by xfz11 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py (hunk 1 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:1094]: WebPubSub also includes Python, Nodejs

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:1094]: Python, Nodejs are alreay added for webapp+webpubsub

this is for Spring-Cloud, spring-cloud itself doesn't support Python NodeJs, right

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**